static/src/javascripts/projects/common/modules/commercial/slice-adverts.js

```diff
@@ -35,27 +35,38 @@ define([
                 return false;
             }
 
-            var container, containerId, $adSlice, isFrontFirst,
+            var container, containerId, $adSlice, isFrontFirst, fabricAdSlot,
                 opts = defaults(
                     options || {},
                     {
                         containerSelector: '.fc-container',
                         sliceSelector: '.js-fc-slice-mpu-candidate'
                     }
                 ),
-                // get all the containers
+            // get all the containers
                 containers   = qwery(opts.containerSelector),
                 index        = 0,
                 adSlices     = [],
                 containerGap = 1,
-                prefs        = userPrefs.get('container-states');
+                prefs        = userPrefs.get('container-states'),
+                addFabricAd  = (config.page.isFront && config.switches.fabricAdverts && detect.isBreakpoint({max : 'phablet'}));
+
+            // insert fabric advert at first container in lieu of a top slot
+            if (addFabricAd) {
+                fabricAdSlot = bonzo(createAdSlot('fabric', 'container-inline'));
+                fabricAdSlot.addClass('ad-slot--mobile');
+                fabricAdSlot.insertAfter(containers[0]);
+            }
+
+            // skip the initial containers if we've already added an advert to the first
+            index+= addFabricAd ? (1 + containerGap) : 0;
```

Comment:
Yeah I guess this way is more obvious, fair enough.

On Mon, 25 Apr 2016 at 15:31 Jimmy Breck-McKye <notifications@github.com>
wrote:

> In
> static/src/javascripts/projects/common/modules/commercial/slice-adverts.js
> <https://github.com/guardian/frontend/pull/12667#discussion_r60922401>:
>
> >                  containers   = qwery(opts.containerSelector),
> >                  index        = 0,
> >                  adSlices     = [],
> >                  containerGap = 1,
> > -                prefs        = userPrefs.get('container-states');
> > +                prefs        = userPrefs.get('container-states'),
> > +                addFabricAd  = (config.page.isFront && config.switches.fabricAdverts && detect.isBreakpoint({max : 'phablet'}));
> > +
> > +            // insert fabric advert at first container in lieu of a top slot
> > +            if (addFabricAd) {
> > +                fabricAdSlot = bonzo(createAdSlot('fabric', 'container-inline'));
> > +                fabricAdSlot.addClass('ad-slot--mobile');
> > +                fabricAdSlot.insertAfter(containers[0]);
> > +            }
> > +
> > +            // skip the initial containers if we've already added an advert to the first
> > +            index+= addFabricAd ? (1 + containerGap) : 0;
>
> Yeah, but doesn't that kinda bury the operation of the loop in another
> block, if you see what I mean?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/guardian/frontend/pull/12667/files/1d0a408044c8211797f93c73a65b51f2bbc674c0#r60922401>
>

-- 

------------------------------
This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396